orangecontrib/educational/widgets/owpolynomialclassification.py

```diff
@@ -1,508 +1,433 @@
-from Orange.widgets.utils import itemmodels
-from math import isnan
-from os import path
 import copy
+from colorsys import rgb_to_hsv, hsv_to_rgb
+from itertools import chain
 
 import numpy as np
-import scipy.sparse as sp
 from scipy.interpolate import splprep, splev
 from scipy.ndimage.filters import gaussian_filter
 
-from AnyQt.QtCore import Qt
-from AnyQt.QtGui import QPixmap, QColor, QIcon
+from AnyQt.QtCore import Qt, QRectF, QObject, QPointF, QEvent
+from AnyQt.QtGui import QPalette, QPen, QFont, QCursor
+from AnyQt.QtWidgets import QGraphicsSceneMouseEvent, QGraphicsTextItem
 
-from Orange.base import Learner as InputLearner
-from Orange.data import (
-    ContinuousVariable, Table, Domain, StringVariable, DiscreteVariable)
-from Orange.widgets import settings, gui
+import pyqtgraph as pg
+
+from Orange.data import \
+    Table, Domain, ContinuousVariable, StringVariable, DiscreteVariable
+from Orange.classification import \
+    Learner, LogisticRegressionLearner, RandomForestLearner, TreeLearner
```

Comment:
Probably it is an issue that needs to be sorted out in Orange, but using `Learner` from `Orange.classification` does not allow `Tree`, `SVM` or 'Random Forest` to be connected to the widget.

Reply:
This is interesting. I now import from `Orange.base`, so it works.